Ginger/GingerCoreTest/Misc/DataSourceTest.cs

```diff
@@ -0,0 +1,264 @@
+
+            //Act : delete colunm
+            accessDataSource.RemoveColumn("MyCustomizedDataTable", "New");
+            ColunmList = accessDataSource.GetColumnList("MyCustomizedDataTable");
+            IsColunmExists = ColunmList.Any(s => s.Contains("New"));
+            //Assert
+            Assert.AreEqual(IsColunmExists, false);
+        }
+
+        [TestMethod]
+        public void RenameTable()
+        {
+            //Arrange
+            string NewName = "TestRename";
+
+            //Act
+            accessDataSource.RenameTable("MyCustomizedDataTable", NewName);
+            Assert.AreEqual(accessDataSource.IsTableExist(NewName), true);
+            accessDataSource.RenameTable(NewName, "MyCustomizedDataTable");
+
+            //Assert
+            Assert.AreEqual(accessDataSource.IsTableExist("MyCustomizedDataTable"), true);
+        }
+
+        [TestMethod]
+        public void SaveTable()
+        {
+            // Arrange
+            ObservableList<DataSourceTable> dataSourceTableList = accessDataSource.GetTablesList();
+            DataSourceTable dataSource = null;
+            List<string> mColumnNames = null;
+            foreach (DataSourceTable dataSourceTable in dataSourceTableList)
+            {
+                if (dataSourceTable.Name == "MyCustomizedDataTable")
+                {
+                    dataSource = dataSourceTable;
+                }
+            }
+            int rowsCount = accessDataSource.GetRowCount(dataSource.Name);
+            DataTable dataTable = accessDataSource.GetQueryOutput("select * from MyCustomizedDataTable");
+            dataSource.DataTable = dataTable;
+            accessDataSource.AddRow(mColumnNames, dataSource);
+
+            //Act
+            accessDataSource.SaveTable(dataSource.DataTable);
+            DataTable res = accessDataSource.GetQueryOutput("select * from MyCustomizedDataTable");
+
+            //Assert
+            Assert.AreEqual(rowsCount + 1, res.Rows.Count, "RowCount");
+        }
+
+        [TestMethod]
+        public void GetTable()
+        {
+            // Arrange
+            string TableName = "MyCustomizedDataTable";
+
+            // Act
+            DataTable dataTable = accessDataSource.GetTable(TableName);
+
+            //Assert
+            Assert.AreEqual(dataTable.TableName, TableName);
+
+        }
+
+        [TestMethod]
+        public void AddRow()
+        {
+            // Arrange
+            ObservableList<DataSourceTable> dataSourceTableList = accessDataSource.GetTablesList();
+            DataSourceTable dataSource = null;
+            List<string> mColumnNames = null;
+            foreach (DataSourceTable dataSourceTable in dataSourceTableList)
+            {
+                if (dataSourceTable.Name == "MyCustomizedDataTable")
+                {
+                    dataSource = dataSourceTable;
+                }
+            }
+            int rowsCount = accessDataSource.GetRowCount(dataSource.Name);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['dataSource' is null on at least one execution path.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4855725)